src/templates/partials/lighttpd.hbs

```diff
@@ -38,19 +16,54 @@ ssl.privkey = "/path/to/private_key"
 ssl.pemfile = "/path/to/signed_cert_followed_by_intermediates"
  {{#if (minver "1.0.2" form.opensslVersion)}}
   {{#if (minver "1.1.0" form.opensslVersion)}}
-ssl.openssl.ssl-conf-cmd = ("MinProtocol" => {{#if (includes "TLSv1" output.protocols)}}"TLSv1"{{else if (includes "TLSv1.1" output.protocols)}}"TLSv1.1"{{else if (includes "TLSv1.2" output.protocols)}}"TLSv1.2"{{else}}"TLSv1.3"{{/if}})
+   {{#if (includes "TLSv1" output.protocols)}}
+ssl.openssl.ssl-conf-cmd = ("MinProtocol" => "TLSv1")
+   {{else if (includes "TLSv1.1" output.protocols)}}
+ssl.openssl.ssl-conf-cmd = ("MinProtocol" => "TLSv1.1")
+   {{else if (includes "TLSv1.2" output.protocols)}}
+    {{#if (minver "1.4.78" form.serverVersion)}}
+ssl.openssl.ssl-conf-cmd = ("MinProtocol" => "TLSv1.2")
+    {{else}}
+     {{#unless (minver "1.4.56" form.serverVersion)}}
+ssl.openssl.ssl-conf-cmd = ("MinProtocol" => "TLSv1.2")
+     {{/unless}}
+    {{/if}}
+   {{else if (includes "TLSv1.3" output.protocols)}}
+    {{#unless (minver "1.4.78" form.serverVersion)}}
+ssl.openssl.ssl-conf-cmd = ("MinProtocol" => "TLSv1.3")
+    {{/unless}}
+   {{/if}}
```

Comment:
(This is one of the extremely complicated ways to "not emit defaults" at any cost, making the actual conditional spaghetti hard to maintain and understandable without some tribal knowledge. I'd much rather just print `output.protocols[0]` as the lowest version /sorted/ and be done. TBD… Might be OK~ish tentatively if addressed once more powerful templating logic is available.)

Reply:
Things will be simpler if ssl-config-generator stopped "recommending" TLS1 and TLSv1.1 (for the Old config) since TLSv1.2 or better is STRONGLY recommended and often required in modern professional settings.

Reply:
That has nothing to with the fact that instead of simply transforming the output object to the config result, this tries to sniff into its content and adjust based on the literal value … for what I would perceive as no real gain. Compared to elsewhere:

https://github.com/mozilla/ssl-config-generator/blob/66667fa52ae1990ab289c9fe49b53fa5b3cf1422/src/templates/partials/postgresql.hbs#L18-L20

🤷

Reply:
@gstrauss Your main concern is forward compatibility by leaving out the defaults. In this case you generate a config for a version N, but just leaving out the defaults at that time, for version N, unknown what may come later. If that config content gets carried over to version N+1, can you still assure it won't break the expectations and recommendation tiers? (=You can't.) So I see the value in being explicit by not having the reality change behind the scenes for the user, as consistency and repeatability is one of the benefits of these configs.

If you configured `intermediate` some time/versions ago (with TLS1.2 min left out as implicit), you'd basically get it disabled and be forced to `modern` unknowingly over time and lighttpd updates once TLS1.2 is not the min default anymore. And that's not a feature, that's definitely a bug — the config won't be stable over time. (And you're aware of  that, it's not an omission but definitely intentional… and I'm not willing to introduce intentional breakage or time bombs just to appease authors' perceptions of their own configs or defaults chosen — which are sort of irrelevant to the output of this tool.)

_(This is borderline unreviewable for me as i can't even be sure the logic is not flawed… spending so much extra time on something i don't think should ever be the scope of the tool here…)_

My opinion is this is doing the consumers disservice, but I don't see any value spending any more time trying to review it responsibly when I see it won't make any difference, so just the usual compromise:

Same as elsewhere, if you want to insist on not setting the value, at least emit it commented out with the note it's the default mentioning version numbers too — so when that changes over time, folks still know what is/was/should have been the explicit value and/or can uncomment it if there's a mismatch in their expectations.
(=Have them yell at server authors for changing things around. Not yell at @mozilla for generating crappy config boilerplates.)

Reply:
To demonstrate what I mean by inconsistent results:

I generate intermediate config for current 1.4.76 and use that — I get intermediate.
I use the same config in 1.4.78 once it comes out — and surprise, I get modern.
I have to come back to generate intermediate again this time for the newer 1.4.78 — and only then get intermediate back.

Basically the claim is the configs are more future proof, but in reality it makes them really compatible only with the version they were generated for — as there's so much logic that tries hard to leverage version-by-version changes to default, that once generated, it really should not be used with different server version, as you get different results than it was generated for.

I don't see it as positive, but at least with the comments stating moz config defaults at time of output and hardcoding the version it was generated for it should not confuse people too much.

Reply:
NB: This changeset already includes the future defaults changing in a version yet unreleased — actually demonstrating the effect of the concerns above right here in the outputs how an `intermediate` today will just stop working for TLSv1.2-only clients once you cross the version numbers hardcoded in the template to emit different configs, diverting from the guidelines in favor of server defaults on purpose again — but if you come back here and punch in your new version number, you'll have this time bomb removed and have a matching config again just fine. Escapes me.

We should try to widen the compatibility with different versions as much as we're able to, not making it as narrow and specific as possible. 🤷

I don't see the value in forcing people to come back to re-generate the template here after every upgrade.